GECKOInstaller.m

```diff
@@ -0,0 +1,43 @@
+classdef GECKOInstaller
+% GECKOInstaller
+%   Support for installing and uninstalling
+%   Run GECKOInstaller.install to install (will set up the path in MATLAB)
+%   Run GECKOInstaller.uninstall to clear the path from MATLAB
+%   To install, you first need to cd to the GECKO root.
+%
+% Johan Gustafsson, 2022-07-07
+%
+    methods (Static)
+        function install
+            sourceDir = fileparts(which(mfilename));
+            paths = GECKOInstaller.GetFilteredSubPaths(sourceDir, '.*\.git.*');
+            addpath(paths);
+            savepath;
+        end
+        function uninstall
+            sourceDir = fileparts(which(mfilename));
+            paths = GECKOInstaller.GetFilteredSubPaths(sourceDir, '.*\.git.*');
+            rmpath(paths);
+            savepath;
+        end
+        function path = getGECKOMainPath()
+			path = fileparts(which(mfilename));
+			path = strrep(path, '\', '/'); %get rid of backslashes in Windows
+			if ~endsWith(path, '/')
+				path = strcat(path,'/');
+			end
+		end
+
+        function newPaths = GetFilteredSubPaths(path_, filter_)
+            % Will fail if you have a directory containing ';'
```

Comment:
It looks like it would fail silently, which would then be tricky to debug. A more obvious failure would instead be easier to sort out.

Reply:
I'm thinking we can leave this as is - it is difficult to fix. The problem is that the matlab function I'm using to get all subdirectories, genpath, will fail in such cases - it returns the directories separated by ';', so such directories will be split in the returned string. There is no easy way to detect it. And, the consequence is small - I would say it is fairly uncommon to have a ';' in a folder name - there are no such in GECKO right now, and let's not add any.

Reply:
`;` are allowed in file paths in some operating systems, so unlikely but still possible.
I was thinking of a simple check if the path [contains](https://www.mathworks.com/help/matlab/ref/contains.html) `;` then exit with error message that `;` are not supported in the file paths.

Reply:
The problem is this: I call a matlab function which returns all subpaths in a directory as a string, where each subpath is separated by semicolon. There is no easy way to tell if a semicolon represents a separator or if that semicolon is part of a directory name. Let's not spend more time on this, it is only a theoretical problem. Remember that the purpose of this function is to install GECKO - and we don't have code in any folders containing semicolons, and it is very unlikely that we will add such, so this is not a real problem.

Reply:
The `;` can also appear one folder above the GECKO installation, so while not a GECKO-caused problem, it would still lead to a problem for the user. The `contains` could thus be applied to the `sourceDir` of the install and uninstall functions, and then I would feel confident in removing the comment that `;` in paths would lead to failures.

Also, looking at the [`genpath` documentation](https://www.mathworks.com/help/matlab/ref/genpath.html) and testing in R2021b, the separator and problematic character is in fact `:` not `;`. Perhaps it would better to define this as a PATH_SEPARATOR, since it is reused in several lines.

Reply:
Ok, fair point, fixed.

Reply:
> Also, looking at the [`genpath` documentation](https://www.mathworks.com/help/matlab/ref/genpath.html) and testing in R2021b, the separator and problematic character is in fact `:` not `;`.

I'm not sure this was (meant to be) addressed? So I added some suggestions below. Therefore I'm resolving the comment thread here to continue the discussion below if necessary.